### PR TITLE
[REVIEW] Update docs to remove references to master [skip ci]

### PR DESCRIPTION
**Pull request by dillon-cullinan**
This PR removes unecessary references to master or updates references to the new main branch in CI scripts and other OPS-owned files.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -75,8 +75,8 @@ UCX
     git checkout v1.8.x
     # apply UCX IB registration cache patches, improves overall
     # IB performance when using a memory pool
-    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch
-    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch
+    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/main/recipe/add-page-alignment.patch
+    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/main/recipe/ib_registration_cache.patch
```
Comment (quasiben) [inline docs/source/install.rst:79]: These won't work until ucx-split-feedstock are moved to `main`

File: docs/source/conf.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #
 # This file does only contain a selection of the most common options. For a
 # full list see the documentation:
-# http://www.sphinx-doc.org/en/master/config
+# http://www.sphinx-doc.org/en/main/config
```
Comment (quasiben) [inline docs/source/conf.py:7]: This is an invalid URL

Comment (quasiben) [reply docs/source/conf.py:7]:
```suggestion
http://www.sphinx-doc.org/en/master/config
```

**Comment by ajschmidt8** [comment]
after addressing @quasiben's comments above, it looks like the only change left is a comment change so this should be safe to merge now.

**Pull request merged**